src/main/java/com/google/starfish/ConnectionPoolContextListener.java

```diff
@@ -0,0 +1,195 @@
+package com.google.starfish;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+import javax.sql.DataSource;
+
+@WebListener("Creates a connection pool that is stored in the Servlet's context for later use.")
+public class ConnectionPoolContextListener implements ServletContextListener {
+
+  // All database credentials are stored in environment variables
+  private static final String CLOUD_SQL_CONNECTION_NAME = System.getenv("CLOUD_SQL_CONNECTION_NAME");
+  private static final String DB_USER = System.getenv("DB_USER");
+  private static final String DB_PASS = System.getenv("DB_PASS");
+  private static final String DB_NAME = System.getenv("DB_NAME");
```

Comment:
So all of these come from the environment, which gets all the env variables by running the script you sent us? How will a deployed server have access to the env? Will we have to deploy it with the same script, somehow?

Reply:
So in the appengine-web.xml file, you can configure the the environment variables on prod instances of your app, so when deployed, the environment will already be configured